app/src/ui/lib/section-filter-list.tsx

```diff
@@ -174,25 +175,32 @@ interface IFilterListState<T extends IFilterListItem> {
 export class SectionFilterList<
   T extends IFilterListItem
 > extends React.Component<ISectionFilterListProps<T>, IFilterListState<T>> {
+  public static getDerivedStateFromProps(
+    props: ISectionFilterListProps<IFilterListItem>,
+    state: IFilterListState<IFilterListItem>
+  ) {
+    return createStateUpdate(props, state)
+  }
+
+  public state: IFilterListState<T> = {
```

Comment:
I went ahead and made this change as well, good point that it infers the type. I added a little bit of logic to adjust the `filterValue` and `filterValueChanged` state during construction. I think it could prevent a re-render, but React is probably smart enough to call `getDerivedStateFromProps` once before rendering for the first time. It might make more sense to use empty values during construction instead. 🤔

Reply:
> React is probably smart enough to call `getDerivedStateFromProps`

Likely so, but, we generally set the state based on the props in the constructor. Seems to make sense to me just for the sake of the state being as accurate as at can be at any given time.